api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java

```diff
@@ -828,20 +828,22 @@ public HL7InArchive getHL7InArchiveByUuid(String uuid) throws APIException {
 	public Message processHL7Message(Message message) throws HL7Exception {
 		// Any post-parsing (pre-routing) processing would go here
 		// or a module can use AOP to do the post-parsing
-		
-		Message response;
+
+		Transportable response;
 		try {
-			if (!router.canProcess(message)) {
```

Comment:
See my comments below first, but some comments on here:

We should maintain something like the `router.canProcess(message)` here. The [original check](https://github.com/hapifhir/hapi-hl7v2/blob/4c5cdded85627978d9553a48f4f2f07cd8d246b9/hapi-base/src/main/java/ca/uhn/hl7v2/app/MessageTypeRouter.java#L65) was fairly simple and we can get close to it with something like this:

```java
private boolean canProcess(Message message) {
	Terser t = new Terser(message);
	ApplicationRouter.AppRoutingData routingData = new AppRoutingDataImpl(t.get("/MSH-9-1"), t.get("/MSH-9-2"), "*", "*");

    return hasActiveBinding(routingData);
}
```

Unfortunately, also, a `Message` object doesn't necessarily implement the `Transportable` interface, so we cannot directly cast either the `message` to a `Transportable` nor the result to a `Message`. Fortunately, translating a `Message` into a `Transportable` shouldn't be hard:

```java
Transportable transportable = new TransportableImpl(message.encode());
```

And, likewise, to convert the `Transportable` result back to a corresponding message:

```java
Message responseMessage = message.getParser().parse(response.getMessage());
```